api/client/joinservice.go

```diff
@@ -125,3 +131,71 @@ func (c *JoinServiceClient) RegisterUsingAzureMethod(ctx context.Context, challe
 	}
 	return certsResp.Certs, nil
 }
+
+// RegisterUsingTPMMethod registers the caller using the TPM join method and
+// returns signed certs to join the cluster. The caller must provide a
+// ChallengeResponseFunc which returns a *proto.RegisterUsingTPMMethodRequest
+// for a given challenge, or an error.
+func (c *JoinServiceClient) RegisterUsingTPMMethod(
+	ctx context.Context,
+	initReq *proto.RegisterUsingTPMMethodInitialRequest,
+	solveChallenge RegisterTPMChallengeResponseFunc,
+) (*proto.Certs, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	tpmJoinClient, err := c.grpcClient.RegisterUsingTPMMethod(ctx)
```

Comment:
Is tpmJoinClient a stream? Maybe:

```suggestion
	stream, err := c.grpcClient.RegisterUsingTPMMethod(ctx)
```

Reply:
Nicer, thank you!
e999ea9928